File: monitor_response.py
from dataclasses import dataclass
from marshalable import Marshalable
import typing as T

@dataclass 
class MonitorResponse(Marshalable):
    """Class to return appropriate error message to client"""
    message: str

    @staticmethod
    def object_type():
        # TODO: Add proper object type
        return 999

    @staticmethod 
    def from_fields(fields: T.Dict[int, T.Any]) -> "Marshalable":
        # TODO: Implement actual construction from fields
        return MonitorResponse(fields[0])
    
    def get_fields(self) -> T.Dict[int, T.Any]:
        # TODO: Implement actual getting of fields
        return {
            0: self.message
        }

    @staticmethod
    def get_field_types() -> T.Dict[int, type]:
        return {
            0: str,
        }

File: client/MonitorResponse.go
package main

import "reflect"

type MonitorResponse struct {
	message string
}

func (mr MonitorResponse) get_fields() map[int]any {
	m := make(map[int]any)
	m[0] = mr.message
	return m
}

func (mr MonitorResponse) from_fields(fields map[int]any) Marshalable{
	//return qMessage;
	return MonitorResponse{fields[0].(string)} 
}

func (mr MonitorResponse) get_field_types() map[int]reflect.Type {
	m := make(map[int]reflect.Type)
	m[0] = reflect.TypeOf(mr.message)
	return m
}

func (mr MonitorResponse) object_type() int {
	return 999
}

func (mr MonitorResponse) extractMssg() string {
	return mr.message
}